File: flexing/src/main/java/net/bradmartin/flexing/Images.java
package net.bradmartin.flexing;

import android.graphics.Bitmap;
import android.graphics.drawable.BitmapDrawable;
import android.widget.ImageView;

public class Images {
    public Bitmap getBitmapFromImageView(ImageView value) {
        // if not instance of imageview throw an exception
        if (!(value instanceof ImageView)) {
            throw new IllegalArgumentException("Value passed to getBitmapFromImageView is not of type ImageView");
        }

        BitmapDrawable drawable = (BitmapDrawable) value.getDrawable();
        return drawable.getBitmap();
    }
}

File: flexing/src/main/java/net/bradmartin/flexing/Images.kt
package net.bradmartin.flexing

import android.graphics.Bitmap
import android.graphics.drawable.BitmapDrawable
import android.widget.ImageView

fun getBitmapFromImageView(value: ImageView): Bitmap {
    // if not instance of ImageView throw an exception
    if (value !is ImageView) {
        throw IllegalArgumentException("Value passed to getBitmapFromImageView is not of type ImageView")
    }

    val drawable = value.drawable as BitmapDrawable
    return drawable.bitmap
}
